cloudinit/sources/DataSourceGPortal.py

```diff
@@ -0,0 +1,97 @@
+# Author: Alexander Birkner <alexander.birkner@g-portal.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit import log as logging
+from cloudinit import sources
+from cloudinit import util
+
+import cloudinit.sources.helpers.gportal as gportal_helper
+
+LOG = logging.getLogger(__name__)
+
+BUILTIN_DS_CONFIG = {
+    'metadata_url': 'http://169.254.169.254/metadata/v1.json',
+}
+
+
+class DataSourceGPortal(sources.DataSource):
+    dsname = 'GPortal'
+
+    # Available server types
+    ServerTypeBareMetal = 'BARE_METAL'
+    ServerTypeVirtual = 'VIRTUAL'
+
+    def __init__(self, sys_cfg, distro, paths):
+        super(DataSourceGPortal, self).__init__(sys_cfg, distro, paths)
```

Comment:
I don't think we need the `super(DataSourceGPortal, self)` part, just `super()` will be enough since we are now only supporting python3